rust/worker/src/log/log.rs

```diff
@@ -202,3 +212,68 @@ impl ChromaError for GetCollectionsWithNewDataError {
         }
     }
 }
+
+#[derive(Clone)]
+pub(crate) struct LogRecord {
+    pub(crate) collection_id: String,
+    pub(crate) log_id: i64,
+    pub(crate) log_id_ts: i64,
+    pub(crate) record: Box<EmbeddingRecord>,
```

Comment:
One thing tricky about this design is you won't be able to move the record out of this to pass it down to lower levels. Do we expect to pass LogRecords around? 

If not, this could be Option<> and we could use take()